cmd/autoheal/object_template.go

```diff
@@ -121,33 +121,55 @@ func (t *ObjectTemplate) Process(object interface{}, data interface{}) error {
 	if glog.V(2) {
 		glog.Infof("Data: %v", data)
 	}
-	return t.processValue(reflect.ValueOf(object), data)
+	v, err := t.processValue(reflect.ValueOf(object), data)
+	if err == nil {
+		object = v.Interface()
+	}
+
+	return err
 }
 
-func (t *ObjectTemplate) processValue(value reflect.Value, data interface{}) error {
-	var err error
-	if value.IsValid() {
-		switch value.Kind() {
+func (t *ObjectTemplate) processValue(inputValue reflect.Value, data interface{}) (val reflect.Value, err error) {
+	val = inputValue
+	if val.IsValid() {
+		switch val.Kind() {
 		case reflect.String:
-			err = t.processString(value, data)
+			text, err := t.processString(val, data)
+			if err == nil {
+				return reflect.ValueOf(text), err
+			}
 		case reflect.Array:
 			// Not implemented yet.
 		case reflect.Slice:
 			// Not implemented yet.
+		case reflect.Map:
+			for _, k := range val.MapKeys() {
+				v, err := t.processValue(val.MapIndex(k), data)
+				if err == nil {
+					// update the processed vaule in the map
+					val.SetMapIndex(k, v)
+				}
+			}
 		case reflect.Struct:
-			for i, n := 0, value.NumField(); i < n && err == nil; i++ {
-				err = t.processValue(value.Field(i), data)
+			for i, n := 0, val.NumField(); i < n && err == nil; i++ {
+				_, err = t.processValue(val.Field(i), data)
```

Comment:
We also need to explicitly check for `err != nil` here. That was incorrect before this pull request already.